tools/setupResourceScripts/setup_tdr_resources.py

```diff
@@ -1,213 +1,278 @@
+        }
+        print(f"Ingesting data into {dataset_to_upload['name']}/{table}")
+        wait_for_job(clients,
+          clients.datasets_api.ingest_dataset(dataset_id, ingest=ingest_request))
+      else:
+        print(f"Skipping ingest of {dataset_to_upload['name']}/{table} because it is empty")
+
+
+def create_billing_profile(clients, add_jade_stewards, cloud_platform, billing_profile_file_name,
+                           azure_managed_app_name):
+  if not billing_profile_file_name:
     billing_profile_file_name = "billing_profile.json"
-    if cloud_platform == "azure":
-        billing_profile_file_name = "azure_billing_profile.json"
-    with open(os.path.join("files", billing_profile_file_name)) as billing_profile_json:
-        billing_profile_request = json.load(billing_profile_json)
-        profile_id = str(uuid.uuid4())
-        billing_profile_request['id'] = profile_id
-        billing_profile_request['profileName'] = billing_profile_request['profileName'] + f'_{profile_id}'
-        print(f"Creating billing profile with id: {profile_id}")
-        profile = wait_for_job(clients, clients.profiles_api.create_profile(billing_profile_request=billing_profile_request))
-        if add_jade_stewards:
-            add_billing_profile_members(clients, profile_id)
-        return profile
+    if cloud_platform=="azure":
+      billing_profile_file_name = "azure_billing_profile.json"
+  with open(os.path.join("files", billing_profile_file_name)) as billing_profile_json:
+    billing_profile_request = json.load(billing_profile_json)
+    billing_profile_id = billing_profile_request['id']
+    billing_profile_request['profileName'] = billing_profile_request[
+                                               'profileName'] + f'_{billing_profile_id}'
+    if azure_managed_app_name:
+      billing_profile_request['applicationDeploymentName'] = azure_managed_app_name
+    print(f"Checking if billing profile with id {billing_profile_id} already exists")
+    try:
+      profile = clients.profiles_api.retrieve_profile(billing_profile_id)
+      print(f"Found profile {billing_profile_id}, reusing")
+      return profile.id
+    except ApiException:
+      print(f"Profile {billing_profile_id} not found")
+    print(f"Creating billing profile with id: {billing_profile_id}")
+    profile = wait_for_job(clients, clients.profiles_api.create_profile(
+      billing_profile_request=billing_profile_request))
+    if add_jade_stewards:
+      add_billing_profile_members(clients, billing_profile_id)
+    return profile['id']
 
 
 def add_billing_profile_members(clients, profile_id):
-    clients.profiles_api.add_profile_policy_member(profile_id,
-                                                   "owner",
-                                                   {'email': 'JadeStewards-dev@dev.test.firecloud.org'})
+  clients.profiles_api.add_profile_policy_member(profile_id,
+    "owner",
+    {
+      'email': 'JadeStewards-dev@dev.test.firecloud.org'})
```

Comment:
Oh interesting - we also set this in the dataset config. Interesting. Maybe we shouldn't add them here as well?  This email won't be valid if we're pointing to a BEE.

Reply:
If its set in the dataset config, I'll just pull it here.

Reply:
Oh, actually, this is the profiles_api, so the dataset wouldn't do it. Maybe we shouldn't set that anyway?

Reply:
Oh, I thought this was adding members to the dataset. 

Interesting.. I think this is needed for the integration test setup (we want to make sure our test users are owners on the billing profile). But, instead of hardcoding this email, I think we could pull from the stewards list on the dataset request and add them as owners on the billing profile as well. 
But, this is getting to be scope creep, so feel free to leave it as is.